### PR TITLE
[Keyboard] Add Runner3680 with 5x6+5x8 layout

### PR DESCRIPTION
**Pull request by Hybrid-Force**
## Description

Adds Runner3680 with 5x6+5x8 layout variant which is a very practical layout. It includes config for RGB matrix feature, basic keymaps which are duplicated and modified from original 5x8 layout, my custom keymap, and via support.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please review the [PR Checklist document](https://docs.qmk.fm/#/pr_checklist), and apply the changes outlined in it.

File: keyboards/runner3680/5x6_5x8/config.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN D3
+#ifdef RGBLIGHT_ENABLE
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/config.h:55]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/runner3680/5x6_5x8/info.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
+{
+    "keyboard_name": "runner3680 5x6+5x8",
+    "url": "",
+    "maintainer": "omkbd",
+    "width": 15,
+    "height": 5,
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/info.json:6]:
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/JIS/keymap.c [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern keymap_config_t keymap_config;
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+rgblight_config_t RGB_current_config;
+#endif
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/JIS/keymap.c:26]:
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/JIS/keymap.c [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+          layer_off(_ADJUST);
+        }
+        break;
+
+      case RGBRST:
+        #ifdef RGBLIGHT_ENABLE
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_config = rgblight_config;
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/JIS/keymap.c:109]: This isn't actually be used and should be removed. 
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+ */
+
+#include QMK_KEYBOARD_H
+
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+rgblight_config_t RGB_current_config;
+#endif
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/default/keymap.c:24]:
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+          layer_off(_ADJUST);
+        }
+        break;
+
+      case RGBRST:
+        #ifdef RGBLIGHT_ENABLE
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_config = rgblight_config;
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/default/keymap.c:107]:
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+ */
+
+#include QMK_KEYBOARD_H
+
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+rgblight_config_t RGB_current_config;
+#endif
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/via/keymap.c:24]:
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+          layer_off(_ADJUST);
+        }
+        break;
+
+      case RGBRST:
+        #ifdef RGBLIGHT_ENABLE
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_config = rgblight_config;
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/via/keymap.c:107]:
```suggestion
```

File: keyboards/runner3680/5x6_5x8/keymaps/derekhsu/rules.mk [hunk 9]
```diff
@@ -0,0 +1,4 @@
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
```
Comment (drashna) [inline keyboards/runner3680/5x6_5x8/keymaps/derekhsu/rules.mk:2]:
```suggestion
```

This should be moved to the keyboard's rules.mk file.

**Review by drashna — CHANGES_REQUESTED** [review]
The `JIS` files need to be renamed so that they're all lower cased.

**Review by tzarc — APPROVED** [review]

**Pull request merged**